scripts/src/main/resources/scripts/command/docker

```diff
@@ -16,46 +16,124 @@ source "$(dirname "${0}")"/../functions
 
 # $1: optional setup
 function doSetup() {  
-  if doIsDockerInstalled 
+  if [ ! "$DOCKER_EDITION" = "DOCKERDESKTOP" ]
   then
-    if [ "${1}" != "silent" ] && ! doIsQuiet
-    then
-      doEcho "docker is already installed at $(command -v docker)"
-      docker version
+    echo "Rancher Desktop set as option"
+    if [[ $(uname -r | grep -i microsoft) ]]
+    then 
+      echo "It seems that you are running the setup from inside a WSL shell."
+      echo "Please use git-bash for that purpose. Otherwise your OS will be interpreted as Linux"
+    else
+      if doIsRancherInstalled 
+      then
+        doIsWindows 
+        queryWin=$?
+        if [ "${1}" != "silent" ] && ! doIsQuiet
+        then
+          #give the User the installation location when using linux or mac
+          if [ queryWin != 0 ]
+          then
+            doEcho "Rancher Desktop is already installed at $(command -v devon docker)"
+          else
+            doEcho "Rancher Desktop is already installed"
+          fi
+        fi
+      else
+        doRequireNotBatch
+        latestversion=$(curl -Ls -o /dev/null -w %{url_effective} https://github.com/rancher-sandbox/rancher-desktop/releases/latest | sed 's/.*\([0-9]\.[0-9]\.[0-9]\).*/\1/')
+        previousversion="1.2.1"
+        if [[ "$latestversion" =~ ^[0-9]\.[0-9].[0-9]$ ]] 
+        then
+          if [ "$latestversion" > "$previousversion" ]
+          then
+            echo "setting RANCHER_VERSION"
+            RANCHER_VERSION=$latestversion
+            echo "RANCHER_VERSION is $RANCHER_VERSION"
+          fi
+        fi
+        local version="${RANCHER_VERSION:-1.2.1}"
+        local code
+        #to download Rancher Desktop we need a bit more work compared to Docker-Desktop 
+        #as the download-URL isn't well structured
+        if doIsWindows
+        then
+          arch="-"
+          ext="windows"
+          os="windows"
+        elif doIsMacOs
+        then
+          ext="mac"
+          os="mac"
+          #options that work with mac OS
+          #arch="arm64"
+          #arch="x86_64"
+          arch="$(uname -m)"
+        else
+          arch="-"
+          ext="other"
+          os="linux"
+        fi
```

Comment:
Isn't there a more elegant solution, because mirrors should take care of everything. Maybe work with code like docker instead. So ideally it's just `doInstall "-" "${installDir}" "rancher" "${version}"` at the end.

Reply:
I would also love that, but the structure of my code is a kind of forced reflection of the urls structure 😆

Reply:
Maybe we can get the changes from @CREITZ25 for `doInstall` and improve this here as well before we merge.
Otherwise, we can also merge as is and create a new story for such improvement.

Reply:
So our conclusion is to wait for this change, merge it into your feature branch and then remove this code-block and use `doInstall` directly instead of `doDownload`?

Reply:
This is now done. Thanks.